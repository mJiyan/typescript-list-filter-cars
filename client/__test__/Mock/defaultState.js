const defaultState = {
    make: 'Ford',
    makes: [
        "ALFA ROMEO",
        "AUDI",
        "AUSTIN",
        "BARKAS",
        "BMW",
        "CADILLAC",
        "CHEVROLET",
        "CHRYSLER",
        "CITROEN",
        "CORVETTE",
        "DACIA",
        "DAEWOO",
        "DAF",
        "DAIHATSU",
        "DODGE",
        "FIAT",
        "FORD",
        "HDPIC",
        "HONDA",
        "HYUNDAI",
        "INFINITI",
        "ISUZU",
        "IVECO",
        "JAGUAR",
        "JEEP",
        "KIA",
        "LADA",
        "LANCIA",
        "LAND ROVER",
        "LANDWIND",
        "LDV",
        "LEXUS",
        "MAZDA",
        "MERCEDES-BENZ",
        "MG",
        "MINI (BMW)",
        "MINI (ROVER)",
        "MITSUBISHI",
        "NISSAN",
        "OPEL",
        "PEUGEOT",
        "PONTIAC",
        "PORSCHE",
        "PROTON",
        "RENAULT",
        "ROVER",
        "SAAB",
        "SEAT",
        "SKODA",
        "SMART",
        "SSANGYONG",
        "SUBARU",
        "SUZUKI",
        "TALBOT",
        "TATA",
        "TESLA",
        "TOYOTA",
        "VOLVO",
        "VW"
    ],
    models: [
        "B-MAX",
        "C-Max",
        "Cougar",
        "Econovan",
        "ECOSPORT",
        "Edge",
        "Escort",
        "Explorer",
        "Fiesta",
        "Fiesta Courier",
        "Focus",
        "Focus C-Max",
        "Fusion",
        "Galaxy",
        "KA",
        "Kuga",
        "Maverick",
        "Mondeo",
        "Mustang",
        "Orion",
        "Probe",
        "Puma",
        "Scorpio",
        "Sierra",
        "S-Max",
        "Streetka",
        "Tourneo Connect",
        "Tourneo Courier",
        "Transit",
        "Transit Connect",
        "Transit Courier",
        "Transit Custom",
        "Windstar"
    ],
    vehicles: [
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 68,
            "enginePowerKW": 50,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1399
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 75,
            "enginePowerKW": 55,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1242
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 80,
            "enginePowerKW": 59,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1388
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 101,
            "enginePowerKW": 74,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1596
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 90,
            "enginePowerKW": 66,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1560
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 50,
            "enginePowerKW": 37,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1119
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 95,
            "enginePowerKW": 70,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1560
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 75,
            "enginePowerKW": 55,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1753
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 54,
            "enginePowerKW": 40,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1097
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 75,
            "enginePowerKW": 55,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1753
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 50,
            "enginePowerKW": 37,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1097
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 131,
            "enginePowerKW": 96,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1796
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 90,
            "enginePowerKW": 66,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1388
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 105,
            "enginePowerKW": 77,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1796
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 73,
            "enginePowerKW": 54,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1391
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 90,
            "enginePowerKW": 66,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1388
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1753
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 103,
            "enginePowerKW": 76,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1598
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 71,
            "enginePowerKW": 52,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1391
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 69,
            "enginePowerKW": 51,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 68,
            "enginePowerKW": 50,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1399
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 75,
            "enginePowerKW": 55,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1242
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 80,
            "enginePowerKW": 59,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1388
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 101,
            "enginePowerKW": 74,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1596
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 90,
            "enginePowerKW": 66,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1560
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 150,
            "enginePowerKW": 110,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1999
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 200,
            "enginePowerKW": 147,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1596
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 69,
            "enginePowerKW": 51,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 88,
            "enginePowerKW": 65,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1597
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 73,
            "enginePowerKW": 54,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1391
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 71,
            "enginePowerKW": 52,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1399
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 54,
            "enginePowerKW": 40,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1119
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 50,
            "enginePowerKW": 37,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1119
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 73,
            "enginePowerKW": 54,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1368
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 103,
            "enginePowerKW": 76,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1596
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 75,
            "enginePowerKW": 55,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1242
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1753
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 50,
            "enginePowerKW": 37,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 75,
            "enginePowerKW": 55,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1242
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 88,
            "enginePowerKW": 65,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1597
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1753
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 101,
            "enginePowerKW": 74,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 998
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 125,
            "enginePowerKW": 92,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 998
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 140,
            "enginePowerKW": 103,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 998
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 49,
            "enginePowerKW": 36,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1087
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 71,
            "enginePowerKW": 52,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1084
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 50,
            "enginePowerKW": 37,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1087
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 45,
            "enginePowerKW": 33,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 935
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 86,
            "enginePowerKW": 63,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1084
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 86,
            "enginePowerKW": 63,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1499
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 101,
            "enginePowerKW": 74,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 998
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 120,
            "enginePowerKW": 88,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1499
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 125,
            "enginePowerKW": 92,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 998
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 105,
            "enginePowerKW": 77,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1596
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 75,
            "enginePowerKW": 55,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1499
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 80,
            "enginePowerKW": 59,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 998
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 97,
            "enginePowerKW": 71,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1567
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 65,
            "enginePowerKW": 48,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 998
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 54,
            "enginePowerKW": 40,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1597
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 69,
            "enginePowerKW": 51,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1271
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 105,
            "enginePowerKW": 77,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1796
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 131,
            "enginePowerKW": 96,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1796
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 50,
            "enginePowerKW": 37,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 182,
            "enginePowerKW": 134,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1596
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 135,
            "enginePowerKW": 99,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1596
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 95,
            "enginePowerKW": 70,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1567
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 75,
            "enginePowerKW": 55,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1368
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 71,
            "enginePowerKW": 52,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1368
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 90,
            "enginePowerKW": 66,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1560
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 150,
            "enginePowerKW": 110,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1999
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 90,
            "enginePowerKW": 66,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1560
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 73,
            "enginePowerKW": 54,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1368
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 69,
            "enginePowerKW": 51,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1271
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 54,
            "enginePowerKW": 40,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1597
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 49,
            "enginePowerKW": 36,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1087
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 97,
            "enginePowerKW": 71,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1567
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 71,
            "enginePowerKW": 52,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1368
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 95,
            "enginePowerKW": 70,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1567
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1242
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 82,
            "enginePowerKW": 60,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1242
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 97,
            "enginePowerKW": 71,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1388
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 120,
            "enginePowerKW": 88,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1596
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 69,
            "enginePowerKW": 51,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 68,
            "enginePowerKW": 50,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1399
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 90,
            "enginePowerKW": 66,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1560
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 140,
            "enginePowerKW": 103,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 998
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 88,
            "enginePowerKW": 65,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1597
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 45,
            "enginePowerKW": 33,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 935
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 71,
            "enginePowerKW": 52,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1391
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1753
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 103,
            "enginePowerKW": 76,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1598
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 103,
            "enginePowerKW": 76,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1598
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 50,
            "enginePowerKW": 37,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1087
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1753
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 73,
            "enginePowerKW": 54,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1391
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 50,
            "enginePowerKW": 37,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1119
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 50,
            "enginePowerKW": 37,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1097
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 105,
            "enginePowerKW": 77,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1796
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 131,
            "enginePowerKW": 96,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1796
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 54,
            "enginePowerKW": 40,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1097
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 71,
            "enginePowerKW": 52,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1391
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 80,
            "enginePowerKW": 59,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1388
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 69,
            "enginePowerKW": 51,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 68,
            "enginePowerKW": 50,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1399
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 75,
            "enginePowerKW": 55,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1368
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 101,
            "enginePowerKW": 74,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1596
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 80,
            "enginePowerKW": 59,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1388
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 75,
            "enginePowerKW": 55,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1560
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 95,
            "enginePowerKW": 70,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1499
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 97,
            "enginePowerKW": 71,
            "fuelType": "Gas",
            "bodyType": "Limousine",
            "engineCapacity": 1388
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 68,
            "enginePowerKW": 50,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1399
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 101,
            "enginePowerKW": 74,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1596
        }
    ]
}

export default defaultState;