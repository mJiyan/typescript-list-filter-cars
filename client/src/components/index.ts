import Header from './Header';
import Card from './Card';
import Filter from './Filter';
import Search from './Search';
import MakeList from './MakeList';
import ModelList from './ModelList';
import CardList from './CardList';


export { Header, Card, Filter, Search, MakeList, ModelList, CardList }