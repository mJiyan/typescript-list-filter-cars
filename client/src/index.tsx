import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import store from './Store';
import App from './App';
import './assets/styles/tailwind.css'

ReactDOM.render(
  <Provider store={store}>
    <App />
  </Provider>,
  document.getElementById('root'),
);
