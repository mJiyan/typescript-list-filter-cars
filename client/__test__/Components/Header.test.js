import React from 'react';
import { Header } from '../../src/components';
import renderer from 'react-test-renderer';

import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { Router, Switch } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { mount, shallow } from 'enzyme';


import defaultState from '../Mock/defaultState';
import toJson from 'enzyme-to-json'; //added this line
import 'jsdom-global/register'; //at the top of file , even  , before importing react



const mockStore = configureMockStore([thunk]);
const store = mockStore(defaultState);
const history = createMemoryHistory()

const wrapper = renderer.create(
  <Provider store={store}>
    <Router history={history}>
      <Switch>
        <Header />
      </Switch>
    </Router>
  </Provider>,
);



const wrapperMount = mount(
  <Provider store={store}>
    <Router history={history}>
      <Switch>
        <Header />
      </Switch>
    </Router>
  </Provider>,
);


describe('rendering header components', () => {
  test('it should match with the snapshot', () => {
    expect(wrapper.toJSON()).toMatchSnapshot();
  });


  test('it should find the darkModeButton', () => {
    const darkButton = wrapperMount.find('[data-testid="dark-mode"]');
    expect(darkButton.length).toEqual(1);
  });

  test('it should make the theme dark', () => {
    const darkButton = wrapperMount.find('[data-testid="dark-mode"]');
    darkButton.simulate('click');
    expect(localStorage.theme).toEqual("dark");
  });

  test('it should make the theme light', () => {
    const darkButton = wrapperMount.find('[data-testid="dark-mode"]');
    darkButton.simulate('click');
    expect(localStorage.theme).toEqual("light");
  });
});

localStorage.setItem('theme', 'dark');
const wrapperDarkThemeMount = mount(
  <Provider store={store}>
    <Router history={history}>
      <Switch>
        <Header />
      </Switch>
    </Router>
  </Provider>,
);


describe('rendering components', () => {
  test('it should make the theme light', () => {
    const darkButton = wrapperDarkThemeMount.find('[data-testid="dark-mode"]');
    darkButton.simulate('click');
    expect(localStorage.theme).toEqual("light");
  });
});