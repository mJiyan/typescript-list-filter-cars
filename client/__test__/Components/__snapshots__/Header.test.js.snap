// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering header components it should match with the snapshot 1`] = `
<nav
  className="bg-WhiteDark dark:bg-DarkBlue h-20 shadow-md"
>
  <div
    className="flex items-center justify-between py-2 h-20 xl:px-20 lg:px-20 md:px-20 sm:px-5 xs:px-3 "
  >
    <span>
      <p
        className="font-bold xl:text-2xl lg:text-2xl md:text-2xl sm:text-md xs:text-sm  tracking-tight "
      >
        Friday Assignment
      </p>
    </span>
    <span
      className=""
    >
      <button
        className="flex flex-cols font-bold text-sm tracking-tight"
        data-testid="dark-mode"
        onClick={[Function]}
      >
        <svg
          className="h-5 w-5"
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20.354 15.354A9 9 0 018.646 3.646 9.003 9.003 0 0012 21a9.003 9.003 0 008.354-5.646z"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={2}
          />
        </svg>
        <p
          className="px-1 font-light"
        >
          Dark Mode
        </p>
      </button>
    </span>
  </div>
</nav>
`;
